src/stream-tcp-reassemble.c

```diff
@@ -1081,12 +1088,15 @@ int StreamTcpReassembleAppLayer (ThreadVars *tv, TcpReassemblyThreadCtx *ra_ctx,
                         "(base %u, isn %u, last_ack %u => diff %u) p %"PRIu64,
                         stream->base_seq, stream->isn, stream->last_ack,
                         stream->last_ack - (stream->isn + ackadd), p->pcap_cnt);
+                gap = stream->last_ack - (stream->isn + ackadd);
```

Comment:
In a preview/test PR #2702 I have removed this seg_list based gap check. I think that makes your PR easier to implement as well.